prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -328,29 +316,62 @@ class FHIRRouter(
-            //  default: must have message id, patient last name, patient first name, dob, specimen type
-            //           must have at least one of patient street, zip code, phone number, email
-            //           must have at least one of order test date, specimen collection date/time, test result date
-            passes = passes &&
-                evaluateFilterCondition(
-                    getQualityFilters(receiver, orgFilters),
+            if (passes) {
+                //  default: must have message id, patient last name, patient first name, dob, specimen type
+                //           must have at least one of patient street, zip code, phone number, email
+                //           must have at least one of order test date, specimen collection date/time, test result date
+                // get the quality filter default result for the bundle, but only if it is needed
+                val qualFilterDefaultResult: Boolean by lazy {
+                    evaluateFilterCondition(
+                        qualityFilterDefault,
+                        bundle,
+                        false,
+                        receiver.reverseTheQualityFilter
+                    )
+                }
+                val filters = getQualityFilters(receiver, orgFilters)
+                passes = evaluateFilterCondition(
+                    filters,
                     bundle,
                     qualFilterDefaultResult,
                     receiver.reverseTheQualityFilter
                 )
+                // log results of filtering things OUT
+                if (!passes) {
+                    logFilterResults(filters, bundle, report, receiver, ReportStreamFilterType.QUALITY_FILTER)
+                }
+            }
+
             // ROUTING FILTER
-            //  default: allowAll
-            passes = passes &&
-                evaluateFilterCondition(getRoutingFilter(receiver, orgFilters), bundle, true)
+            if (passes) {
+                //  default: allowAll
+                val filters = getRoutingFilter(receiver, orgFilters)
+                passes = evaluateFilterCondition(filters, bundle, true)
+
+                // log results of filtering things OUT
+                if (!passes) {
+                    logFilterResults(filters, bundle, report, receiver, ReportStreamFilterType.ROUTING_FILTER)
+                }
+            }
+
             // PROCESSING MODE FILTER
-            //  default: allowAll
-            passes = passes &&
-                evaluateFilterCondition(
-                    getProcessingModeFilter(receiver, orgFilters),
-                    bundle,
-                    processingModeDefaultResult
-                )
+            if (passes) {
+                //  default: allowAll
+                // get the processing mode (processing id) default result for the bundle, but only if it is needed
+                val processingModeDefaultResult: Boolean by lazy {
```

Comment:
Why did you move this?  It is one-per-bundle, no need to calculate for each receiver that passes the previous filters.

Reply:
I moved it just to accompany the other one, which I thought needed to be moved but didn't actually upon review. fixing.....